File: debug_git.js
const fs = require('fs');
const { execSync } = require('child_process');

const logFile = 'git_debug_node.txt';
const commands = [
    'git status',
    'git remote -v',
    'git log -2',
    'git push origin main'
];

let log = '';

commands.forEach(cmd => {
    log += `\nCMD: ${cmd}\n`;
    try {
        const output = execSync(cmd, { encoding: 'utf8', stdio: 'pipe' });
        log += `OUTPUT:\n${output}\n`;
    } catch (e) {
        log += `ERROR:\n${e.message}\n${e.stdout}\n${e.stderr}\n`;
    }
    log += '-'.repeat(20) + '\n';
});

fs.writeFileSync(logFile, log);
console.log('Done writing log');

File: scripts/debug_git.py
import subprocess
import os

def run(cmd):
    try:
        output = subprocess.check_output(cmd, shell=True, stderr=subprocess.STDOUT).decode('utf-8')
        print(f"CMD: {cmd}\nOUTPUT:\n{output}\n{'-'*20}")
    except subprocess.CalledProcessError as e:
        print(f"CMD: {cmd}\nERROR:\n{e.output.decode('utf-8')}\n{'-'*20}")

with open('git_debug_log.txt', 'w') as f:
    import sys
    sys.stdout = f
    run('git status')
    run('git remote -v')
    run('git log -2')
    run('git push origin main')
